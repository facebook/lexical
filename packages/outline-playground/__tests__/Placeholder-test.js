/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import {
  initializeE2E,
  assertHTMLSnapshot,
  assertSelection,
  focusEditor,
} from './utils';

describe('Placeholder', () => {
  initializeE2E({chromium: true, webkit: true, firefox: true}, (e2e) => {
    it(`Displays a placeholder when no content is present`, async () => {
      const {page} = e2e;

      await focusEditor(page);
      const textContent = await page.textContent('.editor-placeholder');
      expect(textContent).toBe('Enter some rich text...');

      await assertHTMLSnapshot(page);
      await assertSelection(page, {
        anchorPath: [1, 0, 0],
        anchorOffset: 0,
        focusPath: [1, 0, 0],
        focusOffset: 0,
      });
    });
  });
});
