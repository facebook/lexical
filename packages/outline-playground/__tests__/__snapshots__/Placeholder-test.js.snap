// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Chromium Displays a placeholder when no content is present 1`] = `"<div contenteditable=\\"false\\" class=\\"editor-placeholder\\">Enter some rich text...</div><p class=\\"editor-paragraph\\"><span>﻿</span></p>"`;

exports[`Placeholder Firefox Displays a placeholder when no content is present 1`] = `"<div class=\\"editor-placeholder\\" contenteditable=\\"false\\">Enter some rich text...</div><p class=\\"editor-paragraph\\"><span>﻿</span></p>"`;

exports[`Placeholder Webkit Displays a placeholder when no content is present 1`] = `"<div contenteditable=\\"false\\" class=\\"editor-placeholder\\">Enter some rich text...</div><p class=\\"editor-paragraph\\"><span>﻿</span></p>"`;
